scipy/spatial/_kdtree.py

```diff
@@ -15,28 +15,39 @@ def minkowski_distance_p(x, y, p=2):
     not extract the pth root. If `p` is 1 or infinity, this is equal to
     the actual L**p distance.
 
+    The last dimensions of `x` and `y` must be the same length.  Any
+    other dimensions must be compatible for broadcasting.
+
     Parameters
     ----------
-    x : (M, K) array_like
+    x : (..., K) array_like
         Input array.
-    y : (N, K) array_like
+    y : (..., K) array_like
         Input array.
     p : float, 1 <= p <= infinity
         Which Minkowski p-norm to use.
 
+    Returns
+    -------
+    dist : ndarray
+        Distance between the input arrays.  Length of the last
```

Comment:
Does it return the distance? I think for L2 it returns the squared distance--I suppose in general it returns the `distance ** p`?